integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/testsupport/types/BigIntegerFieldTypeDescriptor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.tck.testsupport.types;
+
+import java.math.BigInteger;
+import java.util.Optional;
+
+import org.hibernate.search.integrationtest.backend.tck.testsupport.types.expectations.ExistsPredicateExpectations;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.types.expectations.FieldProjectionExpectations;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.types.expectations.FieldSortExpectations;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.types.expectations.IndexNullAsMatchPredicateExpectactions;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.types.expectations.IndexingExpectations;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.types.expectations.MatchPredicateExpectations;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.types.expectations.RangePredicateExpectations;
+
+public class BigIntegerFieldTypeDescriptor extends FieldTypeDescriptor<BigInteger> {
```

Comment:
This is great, but following my comment about encoding big integers as something else than longs, I think there should also be a test for various scales, similar to `DecimalScaleIT`, but for `BigInteger`.

There, you could test various decimal scales, and also test indexing a value that doesn't fit into a long. Those are the most likely to be indexed when people use a `BigInteger`. For example `BigInteger.valueOf( String.valueOf( Long.MAX_VALUE ) + String.valueOf( Long.MAX_VALUE ) )`.